app/models/planning_application_constraint.rb

```diff
@@ -63,8 +64,39 @@ def planning_data_dataset
     planning_data_geojson.dig(:properties, :dataset)
   end
 
+  def sync_consultees!
+    return if consultation.blank? || constraint.blank?
+
+    contacts_for_constraint = constraint.consultee_contacts.where(local_authority_id: local_authority.id)
+
+    consultee_ids = contacts_for_constraint.map { |contact| find_or_create_consultee!(contact).id }.uniq
+
+    existing_consultee_ids = planning_application_constraint_consultees.pluck(:consultee_id)
+    additions = consultee_ids - existing_consultee_ids
+
+    additions.each do |consultee_id|
+      planning_application_constraint_consultees.find_or_create_by!(consultee_id: consultee_id)
+    end
+  end
+
   private
 
+  def find_or_create_consultee!(contact)
+    email = contact.email_address.strip.downcase
+
+    attrs = {
+      email_address: email,
+      name: contact.name,
+      origin: contact.origin,
+      role: contact.role,
+      organisation: contact.organisation
+    }
+
+    consultation.consultees.find_or_create_by!(attrs) do |consultee|
+      consultee.assign_attributes(attrs)
+    end
```

Comment:
I don't think you need to assign the attributes in the block if they're also passing them in to `find_or_create_by`. The block is only needed if you're setting attributes that aren't used to find the object by.